core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -1261,14 +1261,6 @@ public void testAddNewIdentifierFieldColumns() {
   @Test
   public void testAddNestedIdentifierFieldColumns() {
     Schema newSchema = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID)
-        .setIdentifierFields("preferences.feature1")
-        .apply();
-
-    Assert.assertEquals("set existing nested field as identifier should succeed",
```

Comment:
I dont think we should remove this test?

Reply:
This will fail now because `preferences ` is optional, I'll add test for a required struct field.